applications/ShapeOptimizationApplication/custom_utilities/mesh_controller_utilities.h

```diff
@@ -0,0 +1,270 @@
+//
+//  License:         BSD License
+//                   license: ShapeOptimizationApplication/license.txt
+//
+//  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+//
+// ==============================================================================
+
+#ifndef MESH_CONTROLLER_UTILITIES_H
+#define MESH_CONTROLLER_UTILITIES_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "includes/define.h"
+#include "processes/process.h"
+#include "includes/node.h"
+#include "includes/element.h"
+#include "includes/model_part.h"
+#include "includes/kratos_flags.h"
+#include "shape_optimization_application.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+
+*/
+
+class MeshControllerUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of MeshControllerUtilities
+    KRATOS_CLASS_POINTER_DEFINITION(MeshControllerUtilities);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    MeshControllerUtilities( ModelPart& modelPart )
+        : mrModelPart( modelPart )
+    {
+    }
+
+    /// Destructor.
+    virtual ~MeshControllerUtilities()
+    {
+    }
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    // ==============================================================================
+    void UpdateMeshAccordingInputVariable( const Variable<array_1d<double,3>> &rInputVariable )
+    {
+        KRATOS_TRY;
+
+        for(auto & node_i: mrModelPart.Nodes())
+        {
+            array_1d<double,3> variable_value = node_i.FastGetSolutionStepValue(rInputVariable);
+            node_i.X0() += variable_value[0];
+            node_i.Y0() += variable_value[1];
+            node_i.Z0() += variable_value[2];
+            node_i.X() += variable_value[0];
+            node_i.Y() += variable_value[1];
+            node_i.Z() += variable_value[2];            
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    // --------------------------------------------------------------------------
+    void ResetMeshDisplacement()
```

Comment:
Maybe we can find a better name, because this one makes me assume, something happens to the variable ```MESH_DISPLACEMENT```.

Reply:
You are right. Another proposal will follow in the upcoming PR.